packages/labs/analyzer/src/lib/lit-element/events.ts

```diff
@@ -27,22 +32,24 @@ export const getEvents = (node: LitClassDeclaration) => {
         } else if (typeof comment === 'string') {
           const result = parseFiresTagComment(comment);
           if (result === undefined) {
-            // TODO(justinfagnani): report syntax error?
-            continue;
+            throw new DiagnosticsError(
+              tag,
+              'The @fires annotation was not in a recognized form. ' +
+                'Use `@fires event-name {Type} - Description`.'
+            );
           }
           const {name, type, description} = result;
-          // TODO(justinfagnani): how do we dereference the event type name
-          // into a TypeScript type? TypeScript will automatically do this for
-          // jsdoc tags it understands, but unfortunately @fires is not one of
-          // them.
           events.set(name, {
             name,
-            typeString: type,
+            type: type ? programContext.getTypeForJSDocTag(tag) : undefined,
```

Comment:
Why pass the whole tag here and not just the type?

Reply:
The WeakMap holding the pre-processed types for jsdoc tags are keyed off the tag.